### PR TITLE
fix for VS2017 Preview deprecation of ::tr1::tuple

### PR DESCRIPTION
**Pull request by bryanzim**
change static_cast to ImplicitCast_ for consitency
fixes for building with path names containing spaces

### CODE REVIEW

**Comment by bryanzim** [comment]
I signed it!

----------------------------------------

**Comment by bryanzim** [comment]
Can someone review this pull request?

**Comment by bryanzim** [comment]
+1

**Comment by bryanzim** [comment]
Can someone review this with regards to issue #1318.

**Comment by KindDragon** [comment]
In my experience better if you split this PR to 3: ImplicitCast, spaces and tuple warning.

File: googletest/test/gtest-printers_test.cc [hunk 1]
```diff
@@ -1092,8 +1092,8 @@ TEST(PrintTr1TupleTest, VariousSizes) {
   ::std::tr1::tuple<bool, char, short, testing::internal::Int32,  // NOLINT
                     testing::internal::Int64, float, double, const char*, void*,
                     std::string>
-      t10(false, 'a', static_cast<short>(3), 4, 5, 1.5F, -2.5, str, 
-          ImplicitCast_<void*>(NULL), "10");
+      t10(false, 'a', 3, 4, 5, 1.5F, -2.5, str, ImplicitCast_<void*>(NULL),
```
Comment (pavelkryukov) [inline googletest/test/gtest-printers_test.cc]: I would return "ImpicitCast_" here, as some compilers/lints may warn about narrowing conversion

**Comment by bryanzim** [comment]
Thank you for the astute observation.  I have corrected.

**Comment by blizzard4591** [comment]
The updated version of Visual Studio 2017 is now actively being deployed, _please_ merge at least the part for the tr1 usage!

This is a showstopper for builds running with /WX.

**Comment by nbtdev** [comment]
VS2017 15.5 will not build GTest (more specifically, GMock). If someone is waiting for it to be split into three before merging, I will do it (as it seems that @bryanzim has disappeared). 

Note that the fixes in this and in #1343 cannot work on their own, they seem to depend on other changes already in master.

**Comment by bryanzim** [comment]
I am still here just got tired of speaking into a black hole.  This pull requests which has been updated with the latest master builds fine with no warnings or errors using the following CMake command line and the latest Visual Studio 2017.

cmake -G "Visual Studio 15 Win64" -DBUILD_GTEST=ON -DBUILD_GMOCK=ON -Dgtest_build_tests=ON -Dgtest_build_samples=ON -Dgmock_build_tests=ON ..

It used to pass all the unit test, and I think in fact it still does but travis is having problems finding compilers at the moment.

**Comment by nbtdev** [comment]
@bryanzim do we know if no one is looking at it because it's three changes in a single PR? That was the main point of my comment

**Comment by bryanzim** [comment]
At the time I made the changes the implicit_cast and VS2017 changes were both required in order to make VS2017 compile without warnings or errors.

The change for spaces I believe is very simple addition.  I would like to see them all merged at once.  To my knowledge no one has ever looked at this; thus the black hole reference.  I have been working off my own fork since the time of the original pull request.

**Comment by bryanzim** [comment]
Still updating with master and resolving conflicts but no official review or assignee 👎 .

**Comment by bryanzim** [comment]
And again all checks passed same as when initially put forth 2 months ago.

**Pull request closed without merging**